mongodbatlas/resource_mongodbatlas_cloud_provider_access_setup.go

```diff
@@ -0,0 +1,213 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+/*
+	mongodb_atlas_cloud_provider_access_setup
+	-> Creates the the information from the mongodbatlas side
+	-> The delete deletes and deauthorize the role
+*/
+
+func resourceMongoDBAtlasCloudProviderAccessSetup() *schema.Resource {
+	return &schema.Resource{
+		Read:   resourceMongoDBAtlasCloudProviderAccessSetupRead,
+		Create: resourceMongoDBAtlasCloudProviderAccessSetupCreate,
+		Update: resourceMongoDBAtlasCloudProviderAccessAuthorizationPlaceHolder,
+		Delete: resourceMongoDBAtlasCloudProviderAccessSetupDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceMongoDBAtlasCloudProviderAccessSetupImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			// Note: when new providers will be added, this will trigger a recreate
```

Comment:
right, if the cloud provider is change from aws to azure, it will hit a force new, that means to delete the existing one (aws) and send a create for azure

Reply:
Is there any good way to prepare for this to have more than one later?   Or should we just punt that to when (if) it actually comes up?

Reply:
Hi Melissa, when it comes we can just add in the Update request, something like if d.HasChanges("provider_name"), get the old and new value (new accepted providers), in case we need extra logic in case it's needed in an update process.

Reply:
Makes sense.  Thanks for the clarification.